### ISSUE TITLE
"server kick players after restarts"

### ISSUE THREAD

**Issue by trojan-sas** [Southclaw/ScavengeSurvive#267]
its just happen after the server restart however the restart is manual or auto restart it just kick out players whenever they try to login the fix is so annoying as u need to restart the server "twice" to make it works.
LOGS: 
[19:48:01] [LOAD] Snakai (account exists, prompting login) Alive: 1 Last login: Sunday Jun 26 2016 at 19:38:27
[19:48:01] [LOGPROMPT] Snakai is logging in
[19:48:39] [connection] 156.204.5.4:14177 requests connection cookie.
[19:49:10] Kicking 156.204.5.4 because they didn't logon to the game.
[19:49:33] [connection] 156.204.5.4:14212 requests connection cookie.
[19:49:48] [connection] 31.210.187.159:43768 requests connection cookie.
[19:50:09] [connection] 156.204.5.4:14232 requests connection cookie.
[19:50:19] Kicking 31.210.187.159 because they didn't logon to the game.
[19:50:40] Kicking 156.204.5.4 because they didn't logon to the game.

**Comment by Southclaw** [comment]
I just joined and restarted the test server, didn't experience this problem.

Does the kicking happen before or after the world loading sequence? Players can sometimes miss the window between initialisation and the loading sequence if their connection isn't great which results in a connection loss, I've tried to open this window up as much as possible with 500ms gaps between each region loading giving space for the connection to complete: https://github.com/Southclaw/ScavengeSurvive/blob/master/gamemodes/sss/world/world.pwn#L53

**Comment by XFlawless1** [comment]
It's even after it gets loaded I even does not us a filterscript that prevents them. 
We must restart twice

**Comment by Southclaw** [comment]
Could be related to #248 and YSI but as far as I can see, it's not an issue with the core code on its own.

**Comment by trojan-sas** [comment]
i don't think it related to the gap between loading sequence and initialization as it just works fine after we restart the server one more time also it still kicks player even after the world loading is done so i think the real question is what happened in the 2nd. restart not happened in the 1st restart

**Comment by Southclaw** [comment]
Are any of these restarts "hard restarts" (kill process)?
You may benefit from using the `server/crash-on-exit` setting as a workaround for now.

**Comment by trojan-sas** [comment]
"fixed" its related to item array

**Comment by Southclaw** [comment]
[Please make a pull request so we can all benefit from the fix :)](https://github.com/Southclaw/ScavengeSurvive#dont-be-selfish)
